src/content/docs/docs/Old docs/fabric-object-caching.mdx

```diff
@@ -5,111 +5,146 @@ title: Caching of object, properties and beahvior
- */
-fabric.minCacheSideLimit = 256;
+The caching strategy is very simple in Fabric.js and there are few properties on the objects or in the top level configuration to slightly tweak it.
+In practice then this process has a lot of costrains and perfomance pitfalls. Please be sure to read this entire page to
+ understand the basics.
+
+## Tweaks and configuration
+
+### Object configuration
+
+Fabric.js objects have some top level properties to handle caching.
+
+```ts
+myObject.objectCaching = true;
+```
+objectCaching property is the main switch of caching per object.
+When is false caching is skipped, unless is required because of some other constrain ( more on that later )
+
+```ts
+myObject.noScaleCache = true;
 ```
+noScaleCache is a boolean that will stop cache invalidation during a scaling operation with the mouse.
+At the end of the scaling operation the object will be redrawn and a new cache will be created for that scale value.
+
+```ts
+myObject.dirty = true;
+```
+This boolean flags the cache status as old/invalid/dirty and will cause a re-render at the next render cycle.
+As of today there isn't a method to force cache refresh imperatively.
+
+### Classes configuration
+
+On top of those per instance properties there are some that are at class level.
+Each class has a static array of `cacheProperties` as this one below:
+
+```ts
+static cacheProperties = [
+  'fill',
+  'stroke',
+  'strokeWidth',
+  'strokeDashArray',
+  'width',
+  'height',
+  'paintFirst',
+  'strokeUniform',
+  'strokeLineCap',
+  'strokeDashOffset',
+  'strokeLineJoin',
+  'strokeMiterLimit',
+  'backgroundColor',
+  'clipPath',
+];
+```
+
+Being static you will find this array on the class definition like:
+
+```ts
+import { Path } from 'fabric';
+
+console.log(Path.cacheProperties);
+
+// outputs
+['fill', 'stroke', 'strokeWidth', ...]
+```
+This array is used from the `set` method to check on cache invalidation.
+The moment you call:
+```ts
+myObject.set({ fill: 'red', name: 'John' });
+```
+Both keys fill and name get checked against the array and if one is found the object status is set to dirty.
+It is suggested to always use the set method, with the object signature or the simpler one when you want to
+ change properties to your objects to avoid stale caches.
+
+### Fabric.js configuration
+
+Caching has a list of compromises in order to be somehow convenient most of the time.
+A cached canvas can't be too big or it will be too slow to be refreshed for example.
+It can't be too small or it will not trigger certain GPU optmization paths in browsers
+ ( this information may be obsolete with browser changes )
+
+In the Fabric.js configuration object there are 3 values that relate to caching:
+
+```ts
+  /**
+   * Pixel limit for cache canvases. 1Mpx , 4Mpx should be fine.
+   * @since 1.7.14
+   * @type Number
+   * @default
+   */
+  perfLimitSizeTotal = 2097152;
+
+  /**
+   * Pixel limit for cache canvases width or height. IE fixes the maximum at 5000
```

Comment:
Is IE even relevant now,  I am not sure if someone even uses it now, or if can use it now at all

Reply:
> Is IE even relevant now, I am not sure if someone even uses it now, or if can use it now at all

IE is [officially unsupported by Microsoft](https://support.microsoft.com/en-us/windows/internet-explorer-downloads-d49e1f0d-571c-9a7b-d97e-be248806ca70), at least for "public" users. 

> Support for Internet Explorer ended on June 15, 2022. Internet Explorer 11 has been permanently disabled through a Microsoft Edge update on certain versions of Windows 10.
>
> If any site you visit needs Internet Explorer 11, you can reload it with Internet Explorer mode in Microsoft Edge.

There _are_ still businesses and government entities which pay for continued support for Windows 7, but.... probably not a browser the Fabric team really needs to be concerned about. Those same entities can pay for someone to handle IE quirks with respect to Fabric.js, if they really need it, at least IMHO.

Reply:
Internet explorer is unsupported in fabric 6, that note can be reworded to remeber why we chose a number under 5000 as default value.

I m still heavily editing this page